currency/manager.go

```diff
@@ -45,6 +45,18 @@ func (p *PairsManager) GetAssetTypes(enabled bool) asset.Items {
 	return assetTypes
 }
 
+// Clone returns a Deep clone of the pair manager
+// Beware; implementation is not very efficient and should not be used in any hot paths
+func (p *PairsManager) Clone() (PairsManager, error) {
+	var pN PairsManager
+	j, err := json.Marshal(p)
+	if err == nil {
+		err = json.Unmarshal(j, &pN)
+	}
+
+	return pN, err //nolint:govet // copylocks not relevant; We're cloning
+}
+
```

Comment:
I think I'd prefer the removal of this function, see my other comments. If you do keep this, it will need a test

Reply:
Done.
I've punished you by adding 2 new methods instead 😈

Reply:
Alternative is to expose the mutex ... actually I prefer that more.
Imma change it.